packages/hub/src/span.ts

```diff
@@ -1,77 +1,197 @@
-import { Span as SpanInterface } from '@sentry/types';
+import { Span as SpanInterface, SpanContext } from '@sentry/types';
 import { uuid4 } from '@sentry/utils';
 
 export const TRACEPARENT_REGEXP = /^[ \t]*([0-9a-f]{32})?-?([0-9a-f]{16})?-?([01])?[ \t]*$/;
 
 /**
- * Span containg all data about a span
+ * Span contains all data about a span
  */
-export class Span implements SpanInterface {
-  public constructor(
-    private readonly _traceId: string = uuid4(),
-    private readonly _spanId: string = uuid4().substring(16),
-    private _sampled?: boolean,
-    private _parent?: Span,
-  ) {}
+export class Span implements SpanInterface, SpanContext {
+  /**
+   * Trace ID
+   */
+  private readonly _traceId: string = uuid4();
 
   /**
-   * Setter for parent
+   * Span ID
    */
-  public setParent(parent: Span | undefined): this {
-    this._parent = parent;
-    return this;
-  }
+  private readonly _spanId: string = uuid4().substring(16);
+
+  /**
+   * Parent Span ID
+   */
+  private readonly _parentSpanId?: string;
+
+  /**
+   * Has the sampling decision been made?
+   */
+  public readonly sampled?: boolean;
+
+  /**
+   * Timestamp when the span was created.
+   */
+  public readonly startTimestamp: number = new Date().getTime() / 1000;
```

Comment:
It might be a good idea to localize this into a function that generally normalizes timestamps from ms to seconds.